src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -551,3 +551,31 @@ function rrule(::typeof(getproperty), F::T, x::Symbol) where {T <: Cholesky}
     end
     return getproperty(F, x), getproperty_cholesky_pullback
 end
+
+# `det` and `logdet` for `Cholesky`
+function rrule(::typeof(det), C::Cholesky)
+    y = det(C)
+    diagF = _diag_view(C.factors)
+    function det_Cholesky_pullback(ȳ)
+        ΔF = Diagonal(_x_divide_conj_y.(2 * ȳ * conj(y), diagF))
+        ΔC = Tangent{typeof(C)}(; factors=ΔF)
+        return NoTangent(), ΔC
+    end
+    return y, det_Cholesky_pullback
+end
+# compute `x / conj(y)`, handling `x = y = 0`
+function _x_divide_conj_y(x, y)
+    z = x / conj(y)
+    # in our case `iszero(x)` implies `iszero(y)`
+    return iszero(x) ? zero(z) : z
+end
+
+function rrule(::typeof(logdet), C::Cholesky)
+    y = logdet(C)
+    diagF = _diag_view(C.factors)
+    function logdet_Cholesky_pullback(ȳ)
+        ΔC = Tangent{typeof(C)}(; factors=Diagonal((2 * ȳ) ./ conj.(diagF)))
```

Comment:
I was wondering that as well (there are some - now hidden - comments above) but it felt like usually we don't handle such things in a special way if it can only be triggered by specific cotangents but is not an immediate consequence of the inputs. Or do we?

Reply:
I don't think we test this for all the rules we should, but a principle that we seem to be agreed on is that zero (co)tangents should be strong zeros (see e.g. https://github.com/JuliaDiff/ChainRulesCore.jl/pull/551#issuecomment-1129121382).

So in this case if `ȳ==0`, then the cotangent of `factors` should be a zero matrix. Otherwise you end up with cases like `zero(logdet(cholesky(A; check=false)))`, which pulls back a zero cotangent through this rule, injecting `NaN`'s into all downstream cotangents, even though the output is unrelated to the value of `A`.

Reply:
I see, I was not aware of this principle. Would maybe good to add it to the docs and possibly CRTestUtils :slightly_smiling_face: 

I'll update the PR accordingly.

Reply:
Should be fixed in https://github.com/JuliaDiff/ChainRules.jl/pull/613/commits/d831cd49fc8a7669e8e10930bf3214da61ba22dc